clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -194,6 +193,7 @@ private boolean allSubscriptionsEqual(Set<String> allTopics,
                                     otherConsumer, otherMemberGeneration,
                                     tp,
                                     otherMemberGeneration);
+                            allPreviousPartitionsToOwner.put(tp, otherConsumer);
```

Comment:
So in line 161 we directly added the current consumer to the map allPreviousPartitionsToOwner. And if we choose to keep that line we should remove the put in the first two conditions in lines 175 and 180 and we can keep the change you put. 
However, I feel like this would be redundant since we're replacing the otherMember and adding it back if the conditions aren't favorable. My suggestion would be to replace line 161 with get instead of put and then keep the rest of the code as it is.

Reply:
updated, plz check again